arches_querysets/datatypes/resource_types.py

```diff
@@ -0,0 +1,181 @@
+
+
+logger = logging.getLogger(__name__)
+
+
+class ResourceInstanceDataType(datatypes.ResourceInstanceDataType):
+    def transform_value_for_tile(self, value, **kwargs):
+        def from_id_string(uuid_string, graph_id=None):
+            nonlocal kwargs
+            for graph_config in kwargs.get("graphs", []):
+                if graph_id is None or str(graph_id) == graph_config["graphid"]:
+                    break
+            else:
+                graph_config = {}
+            return {
+                "resourceId": uuid_string,
+                "ontologyProperty": graph_config.get("ontologyProperty", ""),
+                "inverseOntologyProperty": graph_config.get(
+                    "inverseOntologyProperty", ""
+                ),
+            }
+
+        try:
+            if isinstance(value, (str, dict)):
+                value = [value]
+                raise TypeError
+            return json.loads(value)
+        except TypeError:
+            if isinstance(value, list):
+                transformed = []
+                for inner in value:
+                    match inner:
+                        case models.ResourceInstance():
+                            transformed.append(
+                                from_id_string(str(inner.pk), inner.graph_id)
+                            )
+                        case uuid.UUID():
+                            # TODO: handle multiple graph configs, requires db?
+                            transformed.append(from_id_string(str(inner)))
+                        case str():
+                            # TODO: handle multiple graph configs, requires db?
+                            transformed.append(from_id_string(inner))
+                        case dict():
+                            transformed.append(from_id_string(inner.get("resource_id")))
+                        case _:
+                            transformed.append(inner)
+                return transformed
+            if isinstance(value, models.ResourceInstance):
+                return [from_id_string(str(value.pk), value.graph_id)]
+            raise
+
+    def to_json(self, tile, node):
+        details = self.get_details(tile, node)
+        return {
+            "@display_value": self.get_display_value(tile, node, details=details),
+            "@details": details,
+        }
+
+    def get_display_value(self, tile, node, *, details=None, **kwargs):
+        if details is None:
+            details = self.get_details(tile, node)
+        return ", ".join([detail["display_value"] for detail in details])
+
+    def to_python(self, value, *, tile, **kwargs):
+        if not (related_resources := self.get_related_resources(tile, value)):
+            return None
+        return related_resources[0]
+
+    def get_related_resources(self, tile, value):
+        if not value:
+            return []
+        related_resources = []
+        try:
+            rxrs = tile.resourceinstance.filtered_from_resxres
```

Comment:
naming nit. Though I'd prefer a longer name I'd settle for one of the shorthand we're already using

Reply:
I hear you, will update -- I think having a version of the "original name" in the filtered/overridden name helps clarity even if the original name isn't perfect.